### PR TITLE
add fayette county pa

### PR DESCRIPTION
**Pull request by aaronpdennis**

### CODE REVIEW

**Comment by aaronpdennis** [comment]
Having trouble debugging this... Data is a geoserver shapefile zip. Using `wget` and `unzip` with the data URL extracts a shapefile, EPSG:3857, with seemingly valid geometry and attributes.

screen shot 2016-10-24 at 7 20 09 pm

Here's the log:

```
2016-08-31 19:37:05,253    DEBUG: Requesting http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=shape-zip with args None
2016-08-31 19:37:21,474    DEBUG: Content-Type says "application/zip" for http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=shape-zip
2016-08-31 19:37:21,474    DEBUG: Content-Disposition agrees: "ParcelsPolygon.zip"
2016-08-31 19:37:21,475    DEBUG: Guessed us--pa--fayette-4e137c78.zip for http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=shape-zip
2016-08-31 19:37:21,475    DEBUG: Requesting http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=shape-zip with args None
2016-08-31 19:37:46,129     INFO: Downloaded 11968603 bytes for file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/cache-Hn1QqP/http/us--pa--fayette-4e137c78.zip
2016-08-31 19:37:46,210     INFO: Cached data in file:///tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/cached/us--pa--fayette-4e137c78.zip
2016-08-31 19:37:46,249    DEBUG: URL says ".zip" for file:///tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/cached/us--pa--fayette-4e137c78.zip
2016-08-31 19:37:46,249    DEBUG: Guessed us--pa--fayette-100fbbec.zip for file:///tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/cached/us--pa--fayette-4e137c78.zip
2016-08-31 19:37:46,271    DEBUG: File exists /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/http/us--pa--fayette-100fbbec.zip
2016-08-31 19:37:46,271     INFO: Downloaded to [u'/tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/http/us--pa--fayette-100fbbec.zip']
2016-08-31 19:37:46,923    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.shp
2016-08-31 19:37:46,923    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/wfsrequest.txt
2016-08-31 19:37:46,923    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.cst
2016-08-31 19:37:46,923    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.shx
2016-08-31 19:37:46,924    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.dbf
2016-08-31 19:37:46,924    DEBUG: Expanded file /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.prj
2016-08-31 19:37:46,924     INFO: Decompressed to 6 files
2016-08-31 19:37:47,290     INFO: Sampled 6 records
2016-08-31 19:37:47,290    DEBUG: Converting to /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf
2016-08-31 19:37:47,291    DEBUG: Selected /tmp/worker-REGa9M/work-s2hF7p/out/process_one-yZ4j4s/conform-Qg4Ajf/unzipped/ParcelsPolygon.shp for source
2016-08-31 19:37:47,291    DEBUG: extract temp file /tmp/openaddr-extracted-R0mqOQ.csv
2016-08-31 19:37:47,292     INFO: Converting a layer to CSV: ParcelsPolygon
2016-08-31 19:37:47,292  WARNING: No encoding given and OGR couldn't guess. Trying ISO-8859-1, YOLO!
2016-08-31 19:37:47,292    DEBUG: Assuming shapefile data is encoded iso-8859-1
2016-08-31 19:37:47,298  WARNING: Error doing conform; skipping
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/openaddr/__init__.py", line 185, in conform
    csv_path, addr_count = task4.convert(data, decompressed_paths, workdir)
  File "/usr/local/lib/python2.7/dist-packages/openaddr/conform.py", line 501, in convert
    rc = conform_cli(source_definition, source_path, dest_path)
  File "/usr/local/lib/python2.7/dist-packages/openaddr/conform.py", line 1033, in conform_cli
    extract_to_source_csv(source_definition, source_path, extract_path)
  File "/usr/local/lib/python2.7/dist-packages/openaddr/conform.py", line 980, in extract_to_source_csv
    ogr_source_to_csv(source_definition, ogr_source_path, extract_path)
  File "/usr/local/lib/python2.7/dist-packages/openaddr/conform.py", line 624, in ogr_source_to_csv
    geom.Transform(coordTransform)
  File "/usr/lib/python2.7/dist-packages/osgeo/ogr.py", line 4875, in Transform
    return _ogr.Geometry_Transform(self, *args)
TypeError: in method 'Geometry_Transform', argument 2 of type 'OSRCoordinateTransformationShadow *'
2016-08-31 19:37:47,568  WARNING: Nothing processed
```

**Comment by iandees** [comment]
Rebuild this please.

**Comment by iandees** [comment]
Rerun, this please

**Comment by migurski** [comment]
```python
# Regexp for a PR comment that requests a re-run.
RETEST_COMMENT_PAT = re.compile(r'^re-?run this,? please\b', re.IGNORECASE|re.MULTILINE)
```

Re-run this, please.
Rerun this please!
Rerun this, please.

**Comment by iandees** [comment]
It would appear that PROJ or GDAL is having a hard time reading this projection.

**Comment by migurski** [comment]
Should be possible to [add a custom SRS to the source](https://github.com/openaddresses/openaddresses/blob/master/CONTRIBUTING.md#processing-tags).

**Comment by iandees** [comment]
The tricky bit is that the source comes with a .prj file [0] so machine [will ignore the `srs` attribute](https://github.com/openaddresses/machine/blob/master/openaddr/conform.py#L567).

**Comment by migurski** [comment]
Hm, you're right.

**Comment by aaronpdennis** [comment]
Hi! A little confused. What should be done here?

**Comment by aaronpdennis** [comment]
Geoserver has several different `outputFormat` options. Currently, the data source is calling for the shapefile. Would CSV or JSON work better? 
http://docs.geoserver.org/latest/en/user/services/wfs/outputformats.html

**Comment by iandees** [comment]
I'm not sure what's going on with the Shapefile, @aaronpdennis. Let's try the GeoJSON output and see what happens?

**Comment by iandees** [comment]
The error from gdal is:

```
Error Number: 6
Error Type: Failure
Error Message: No translation for Popular Visualisation Pseudo Mercator to PROJ.4 format is known.
```

Sounds like it doesn't know how to read the .prj file.

File: sources/us/pa/fayette.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "US Census": {
+            "geoid": "42051",
+            "name": "Fayette County",
+            "state": "Pennsylvania"
+        },
+        "country": "us",
+        "state": "pa",
+        "city": "Fayette"
+    },
+    "data": "http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=shape-zip",
```
Comment (iandees) [inline sources/us/pa/fayette.json]: Switch the URL to: 

```
http://fayettemaps.rba.com:8082/geoserver/ows?service=WFS&version=1.0.0&request=GetFeature&typename=fayette:ParcelsPolygon&outputFormat=json
```

File: sources/us/pa/fayette.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "type": "http",
+    "conform": {
+        "type": "shapefile-polygon",
```
Comment (iandees) [inline sources/us/pa/fayette.json]: Switch to `geojson`

File: sources/us/pa/fayette.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "number": "ADDR_PHYSI",
+        "street": [
+            "ADDR_PHY_1",
+            "ADDR_PHY_2"
+        ],
+        "city": "CITYNAME",
+        "postcode": "ZIP1",
+        "region": "STATE"
+    },
+    "compression": "zip"
```
Comment (iandees) [inline sources/us/pa/fayette.json]: Remove `compression` key.

**Comment by iandees** [comment]
Great, thanks!

**Comment by migurski** [comment]
screen shot 2016-10-25 at 1 49 43 pm

**Pull request merged**